src/context-parser-handlebars.js

```diff
@@ -404,6 +430,49 @@ ContextParserHandlebars.prototype.analyzeAst = function(ast, contextParser, char
 
                 output += t.output;
 
+            // if the 'partial expression' is not in Data State,
+            // TODO: we support basic partial only, need to enhance it.
+            // http://handlebarsjs.com/partials.html
+            } else if (node.type === handlebarsUtils.PARTIAL_EXPRESSION && 
+                parser.getCurrentState() !== stateMachine.State.STATE_DATA && 
+                (re = handlebarsUtils.isValidExpression(node.content, 0, node.type)) && 
+                (partialName = re.tag)) {
+
+                key = this._config._namespace === ''? partialName : this._config._namespace + '/' + partialName;
+                partialContent = this._config._partialsCache[key];
+                if (partialContent === undefined) {
+                    msg = "[ERROR] SecureHandlebars: Fail to read the " + node.content + " partial template! ";
+                    exceptionObj = new ContextParserHandlebarsException(msg, this._fileName, this._lineNo, this._charNo);
+                    handlebarsUtils.handleError(exceptionObj, true);
+                }
+
+                // get the html state number right the parital is called, and analyzed
+                enterState = parser.getCurrentState();
+
+                // while analyzing the partial, use the current parser (possibly forked) and disable printChar
+                partialContent = this.analyzeContext(partialContent, {
+                    contextParser: parser,
+                    disablePrintChar: true
+                });
+
+                if (this._config._enablePartialCombine) {
+                    output += partialContent;
+                } else {
+
+                    // https://github.com/yahoo/secure-handlebars/blob/master/src/handlebars-utils.js#L76
+                    var pattern = /^(\{\{~?>\s*)([^\s!"#%&'\(\)\*\+,\.\/;<=>@\[\\\]\^`\{\|\}\~]+)(.*)/,
+                        newPartialName = 'SJST'+enterState+partialName;
```

Comment:
pls add a delimiter
imagine 'SJST'+'1'+'2abc'. it will be considered as in a different enterState